### PR TITLE
feat/tailwind in dockerfile

### PR DESCRIPTION
**Pull request by shuresm57**
- **added curl command to the dockerfile to download tailwind**
- **removed musl from cli binary in dockerfile**
- **moved the dockerfile from ruby-app/ruby-docs to /documentation W**
- **changed the makefile for development**
- **added a how to run in the readme**

### What has changed?

Tailwind CSS is now in the buildstage of the Dockerfile, instead of loaded via CDN.
Also changed the Makefile, so that is how we run it from now on.

### Why did it need to be changed?

The CDN is not meant for production.

### How did you change it?

Downloaded Tailwind v3 CLI in the Dockerfile build stage, added `input.css` + `tailwind.config.js`, and replaced the CDN `<script>` with
   a `<link>` to the compiled `output.css`. Also updated the Makefile with Docker targets and the README with run instructions.

***

**Checklist**

- [x] Application compiles
- [x] Documentation added

### Commit atomicity ⚠️
This PR touches multiple concerns (Docker build, Makefile, docs, Tailwind assets/config, template). Consider splitting into focused commits/PRs for easier review and bisectability:
- docs (documentation/docker.md, readme.md)
- Docker build changes (ruby-app/Dockerfile, move to /documentation)
- Build tooling (ruby-app/Makefile)
- Tailwind assets/config (ruby-app/public/input.css, ruby-app/tailwind.config.js)
- Template change (ruby-app/views/layout.erb)

### Security / Supply-chain
- Good: Dockerfile verifies the Tailwind CLI with a hardcoded SHA-256 — strong practice.
- Issue: documentation/docker.md and README examples still show downloading Tailwind without checksum. Docs should mirror the secure Dockerfile to avoid teaching an insecure workflow.

Suggested doc change (documentation/docker.md) — add checksum verification:
```suggestion
*** documentation/docker.md
@@
-RUN curl -sLO https://github.com/tailwindlabs/tailwindcss/releases/download/v3.4.17/tailwindcss-linux-x64 \
-    && chmod +x tailwindcss-linux-x64 \
-    && ./tailwindcss-linux-x64 -i ./public/input.css -o ./public/output.css --minify
+RUN curl -sLO https://github.com/tailwindlabs/tailwindcss/releases/download/v3.4.17/tailwindcss-linux-x64 \
+    && echo "7d24f7fa191d2193b78cd5f5a42a6093e14409521908529f42d80b11fde1f1d4  tailwindcss-linux-x64" | sha256sum -c - \
+    && chmod +x tailwindcss-linux-x64 \
+    && ./tailwindcss-linux-x64 -i ./public/input.css -o ./public/output.css --minify
```

### Docker image / Automation
- Remove curl from the runtime image to shrink attack surface — curl is only needed in build stage.
Suggested change (ruby-app/Dockerfile):
```suggestion
*** ruby-app/Dockerfile
@@
-RUN apk add --no-cache sqlite-dev tzdata curl \
+RUN apk add --no-cache sqlite-dev tzdata \
     && adduser -D appuser
```
- Use `apk add --no-cache` when installing curl in the build stage to avoid caching index data:
```suggestion
*** ruby-app/Dockerfile
@@
-RUN apk add curl
+RUN apk add --no-cache curl
```
- Port mismatch: Dockerfile EXPOSEs 4567 while app runs on 8080. Align EXPOSE with runtime port to avoid confusion:
```suggestion
*** ruby-app/Dockerfile
@@
-EXPOSE 4567
+EXPOSE 8080
```

### Documentation / Developer Experience (Sharing)
- Makefile `up` target references `--env-file .env` — good that a .env.example was added, but README should enumerate required env vars and reference ruby-app/.env.example. Ensure README and Makefile paths reflect that the Dockerfile moved to /documentation.
- Update documentation examples to match the moved Dockerfile path and final workflow.

### Tests / Edge cases (Measurement / Automation)
- Ensure Tailwind `content` paths (./views/**/*.erb) cover all places classes appear (JS templates or other assets). Missing paths can cause missing styles in production.
- Add CI step or smoke test that builds the Docker image and verifies /public/output.css (or /output.css) exists and contains expected CSS classes; this catches build regressions.

### Positive notes (Culture / Lean)
- Compiling Tailwind during Docker build improves reproducibility and removes CDN runtime dependency — aligns with Automation and Lean.
- Multi-stage build and running as non-root are good security and image-size practices.

### Minor suggestions
- Confirm the CLI binary build variant (musl removal) matches target base image libc to avoid runtime incompatibility.
- Consider documenting why Tailwind CLI version was pinned and add a short upgrade process in docs.

### CODE REVIEW

File: ruby-app/public/input.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
```
[the remark this answers is not in the document]
Comment (shuresm57) [reply ruby-app/public/input.css:3]: We dont use stylelint in our project.
